bionic/provider.py

```diff
@@ -96,6 +98,14 @@ def protocol_for_name(self, name):
                 % (name, tuple(self.attrs.names)))
         return self.attrs.protocols[name_ix]
 
+    @property
```

Comment:
The other attributes are accessed directly (like `provider.attrs.names`, etc.) so for consistency I'd be inclined to do the same here.